metadata-service/auth-impl/src/main/java/com/datahub/authorization/DataHubAuthorizer.java

```diff
@@ -184,15 +189,32 @@ private boolean isRequestGranted(final DataHubPolicyInfo policy, final Authoriza
     if (AuthorizationMode.ALLOW_ALL.equals(mode())) {
       return true;
     }
+
+    Optional<Urn> actorUrn = getUrnFromRequestActor(request.getActorUrn());
+    if (actorUrn.isEmpty()) {
+      return false;
+    }
+
+    final ResolvedResourceSpec actorResolvedResourceSpec = _resourceSpecResolver.resolve(
+            new ResourceSpec(actorUrn.get().getEntityType(), request.getActorUrn()));
     final PolicyEngine.PolicyEvaluationResult result = _policyEngine.evaluatePolicy(
         policy,
-        request.getActorUrn(),
+        actorResolvedResourceSpec,
         request.getPrivilege(),
         resourceSpec
     );
     return result.isGranted();
   }
 
+  private Optional<Urn> getUrnFromRequestActor(String actor) {
```

Comment:
This is fine - let's go with this. 

I do think in future we will want to make the auth layer a bit more opinionated and work in the URN types instead of raw strings. That was probably a bad call initially